engine/dbscan/src/main/java/org/opennms/alec/engine/dbscan/DBScanEngineFactory.java

```diff
@@ -46,7 +68,23 @@ public String getName() {
 
     @Override
     public AbstractClusterEngine createEngine(MetricRegistry metrics) {
-        return new DBScanEngine(metrics, epsilon, alpha, beta);
+        Optional<DistanceMeasureFactory> factory = distanceMeasureFactoryMap.entrySet()
```

Comment:
Is it possible to just do `Optional.ofNullable(distanceMeasureFactoryMap.get(distanceMeasureFactoryName))`?